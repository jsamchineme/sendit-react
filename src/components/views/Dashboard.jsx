import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router-dom';
import MobileHeader from '../common/MobileHeader';
import MainPageHeader from '../common/MainPageHeader';
import MakeOrder from './MakeOrder';
import Sidebar from '../common/Sidebar';
import { toggleMobileMenu } from '../../redux/actions/elements';

class Dashboard extends Component {

  render() {
    let {
      menuActiveOverlayStatus,
      mobileMenuOpen,
      toggleMobileMenu
    } = this.props;

    return (
      <div>
        <div className="wrapper">
          <section className="page-content">
            <MobileHeader />
            <div className="row">
              <div className="column col-3 page-left">
                <Sidebar />
              </div>
              <div className="column col-9 page-right">
                <MainPageHeader />
                <div className="main-content">
                  <div className="container">
                    <Switch>
                      <Route path='/dashboard/make-order' component={MakeOrder} />
                    </Switch>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div className="hide-for-large">
          <div id="menu-active-overlay"
              onClick={() => toggleMobileMenu(mobileMenuOpen)}
            className={menuActiveOverlayStatus} />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    menuActiveOverlayStatus: state.elementStatuses.menuActiveOverlayStatus,
    mobileMenuOpen: state.elementStatuses.mobileMenuOpen,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    toggleMobileMenu: (visibility) => dispatch(toggleMobileMenu(visibility)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);