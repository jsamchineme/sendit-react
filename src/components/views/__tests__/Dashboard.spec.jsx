import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import Dashboard from '../Dashboard';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for login form', () => {
  it('should render the login form', () => {
    const subject = (
      <Provider store={store}>
        <BrowserRouter>
          <Route render={() => <Dashboard />} />
        </BrowserRouter>
      </Provider>
    );
    const wrapper = mount(subject);
  
    // expect(wrapper.state('email')).toBe(value);
    expect(wrapper).toMatchSnapshot();
  });
});