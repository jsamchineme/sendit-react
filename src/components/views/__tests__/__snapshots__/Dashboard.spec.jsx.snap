// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for login form should render the login form 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "auth": Object {
          "isAuthenticated": false,
          "isProcessing": false,
        },
        "elementStatuses": Object {
          "menuActiveOverlayStatus": "",
          "mobileMenuOpen": false,
        },
        "form": Object {},
        "parcels": Array [],
        "serverErrors": Object {
          "userLogin": "",
          "userSignup": "",
        },
      },
    }
  }
>
  <BrowserRouter>
    <Route
      render={[Function]}
    />
  </BrowserRouter>
</ContextProvider>
`;
